### PR TITLE
Tracking Implant shows location to sec on removal

### PR DESCRIPTION
**Pull request by Kyle2143**
## About the PR 
Tracking implants now alert security when they are removed from a person with their location.

## Why's this needed? 
Tracking implants have always been a part of security's toolkit to keep track of dangerous prisoners who were released, but they were seldom used (in my experience) because they could be removed rather easily. 

By alerting security to when one has been removed, it should make the implants more likely to be used so you can use it with assurance that you'll be able to locate a prisoner or at least find out where they _were_ when they removed it so you can track them down. 

## Changelog

```changelog
(u)Kyle:
(*)Tracking implants now alert security when they are removed from a person with their location.
```

### CODE REVIEW

**Comment by Kyle2143** [comment]
Oh, I didn't test this just yet. Just made the pr before going to bed. Should be fine though. But don't merge it yet unless you wanna test it yourself. I'll test and merge tomorrow probably.

File: code/obj/item/implant.dm [hunk 1]
```diff
@@ -28,6 +28,29 @@ THROWING DARTS
...
+	var/datum/radio_frequency/radio_connection
+
+	New()
+		..()
+		if (uses_radio)
+			SPAWN_DBG(10 SECONDS)
+				if (radio_controller)
+					radio_connection = radio_controller.add_object(src, "[frequency]")
+				if (!src.net_id)
+					src.net_id = generate_net_id(src)
```
Comment (ZeWaka) [inline code/obj/item/implant.dm:45]:
```suggestion
					src.net_id = generate_net_id(src)

```

File: code/obj/item/implant.dm [hunk 2]
```diff
@@ -390,6 +396,15 @@ THROWING DARTS
 		STOP_TRACKING
 		..()
 
+	on_remove(var/mob/M)
+		if (!src.owner)
+			return
+		var/coords = src.get_coords()
+		var/message = "TRACKING IMPLANT LOST: [src.owner][coords] in [get_area(src)], "
```
Comment (ZeWaka) [inline code/obj/item/implant.dm:403]: why even have the var(s) at this point?

File: code/obj/item/implant.dm (hunk 1 above)
Comment (Kyle2143) [reply code/obj/item/implant.dm:45]: what?

File: code/obj/item/implant.dm (hunk 2 above)
Comment (Kyle2143) [reply code/obj/item/implant.dm:403]: Copy paste strikes again.

File: code/obj/item/implant.dm (hunk 1 above)
Comment (ZeWaka) [reply code/obj/item/implant.dm:45]: spacing :)

File: _std/defines/jobs.dm [hunk 3]
```diff
@@ -39,4 +39,5 @@
 #define MGA_SHIPPING "Shipping Alert"
 #define MGA_CARGOREQUEST "Cargo Request"
 #define MGA_CRISIS "Crisis Alert"
-#define MGA_RADIO "Radio Alert"
\ No newline at end of file
+#define MGA_RADIO "Radio Alert"
+#define MGA_TRACKING "Tracking Alert"
```
Comment (Adharainspace) [inline _std/defines/jobs.dm:43]: the only things that were in _std seem good so my approving review is now done :smile: (idk enough abt this code to actually review it)

**Review by Adharainspace — APPROVED** [review]

**Pull request merged**